src/screens/nano/NanoContractExecuteMethod.js

```diff
@@ -0,0 +1,635 @@
+    // Depending on the token, the input for the amount changes because it might be an NFT
+    const nft = isNFT(token);
+    let inputNumberProps;
+    if (nft) {
+      inputNumberProps = {
+        placeholder: '0',
+        precision: 0,
+      };
+    } else {
+      inputNumberProps = {
+        placeholder: hathorLib.numberUtils.prettyValue(0, decimalPlaces)
+      };
+    }
+
+    const renderCommon = () => {
+      return (
+        <div className="d-flex flex-grow-1">
+          <div className="col-8">
+            <label>{t`Token`}</label>
+            {renderTokenSelect(null, true, (e) => onActionValueChange(index, 'token', e.target.value))}
+          </div>
+          <div className="col-4">
+            <label>{t`Amount`}</label>
+            <InputNumber
+              required
+              requirePositive={true}
+              onValueChange={amount => onActionValueChange(index, 'amount', amount)}
+              className="form-control output-value"
+              {...inputNumberProps}
+            />
+          </div>
+        </div>
+      );
+    };
+
+    const renderRemoveButton = () => {
+      return (
+        <div className="align-items-end d-flex">
+          <button
+            type="button"
+            className="form-control text-danger remove-action-btn"
+            onClick={() => removeAction(index)}
+          >
+            {t`Remove`}
+          </button>
+        </div>
+      )
+    }
+
+    if (type === NanoContractActionType.WITHDRAWAL) {
+      return (
+        <>
+          <div className="d-flex flex-grow-1">
+            {renderCommon()}
+            {renderRemoveButton()}
+          </div>
+          <div className="d-flex w-50 mt-3">
+            <div className="d-flex flex-column flex-grow-1">
+              <label>{t`Address`}</label>
+              <input
+                required
+                type="text"
+                ref={ref => {
+                  actionAddressesRef.current[index] = ref
+                }}
+                className="form-control"
+                onChange={e => onActionValueChange(index, 'address', e.target.value)}
+              />
+            </div>
+          </div>
+        </>
+      );
+    }
+
+    return (
+      <div className="d-flex flex-grow-1">
+        {renderCommon()}
+        {renderRemoveButton()}
+      </div>
+    );
+  }
+
+  /**
+   * Method executed when user clicks the link to add a new action
+   * It adds an empty action to the actions list
+   *
+   * @param {Event} e Event triggered by the anchor click
+   */
+  const addAction = (e) => {
+    e.preventDefault();
+    actions.push({});
+    setActions([...actions]);
+  }
+
+  /**
+   * Method executed when user clicks the button to remove an action
+   * It removes the action from the list and updates the state
+   *
+   * @param {number} index Index of the action to be removed
+   */
+  const removeAction = (index) => {
+    pullAt(actions, [index]);
```

Comment:
suggestion: Same here, avoid manipulating the state variable directly.

Reply:
Done https://github.com/HathorNetwork/hathor-wallet/pull/627/commits/53b75d624e99b2ab1afa9fe07220ad3d2f3c318b